traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -359,267 +737,482 @@ WHERE s.id IN (` + edgeIDs + `)))
 	}
+	// If type is changing ensure it isn't assigned to any DSes.
+	if typeID != *server.TypeID {
+		if len(dsIDs) != 0 {
+			return errors.New("server type can not be updated when it is currently assigned to Delivery Services"), nil, http.StatusConflict
+		}
+	}
 	// Check to see if the user is trying to change the CDN of a server, which is already linked with a DS
-	if cdnId != *s.CDNID && len(dsIDs) != 0 {
+	if cdnID != *server.CDNID && len(dsIDs) != 0 {
 		return errors.New("server cdn can not be updated when it is currently assigned to delivery services"), nil, http.StatusConflict
 	}
-	// If type is changing ensure it isn't assigned to any DSes.
-	if typeID != *s.TypeID {
-		if len(dsIDs) != 0 {
-			return errors.New("server type can not be updated when it is currently assigned to delivery services"), nil, http.StatusConflict
+
+	return nil, nil, http.StatusOK
+}
+
+func createInterfaces(id int, interfaces []tc.ServerInterfaceInfo, tx *sql.Tx) (error, error, int) {
+	ifaceQry := `
+	INSERT INTO interface (
+		max_bandwidth,
+		monitor,
+		mtu,
+		name,
+		server
+	) VALUES
+	`
+	ipQry := `
+	INSERT INTO ip_address (
+		address,
+		gateway,
+		interface,
+		server,
+		service_address
+	) VALUES
+	`
+
+	ifaceQueryParts := make([]string, 0, len(interfaces))
+	ipQueryParts := make([]string, 0, len(interfaces))
+	ifaceArgs := make([]interface{}, 0, len(interfaces))
+	ipArgs := make([]interface{}, 0, len(interfaces))
+	for i, iface := range interfaces {
+		argStart := i * 5
+		ifaceQueryParts = append(ifaceQueryParts, fmt.Sprintf("($%d, $%d, $%d, $%d, $%d)", argStart+1, argStart+2, argStart+3, argStart+4, argStart+5))
+		ifaceArgs = append(ifaceArgs, iface.MaxBandwidth, iface.Monitor, iface.MTU, iface.Name, id)
+		for _, ip := range iface.IPAddresses {
+			argStart = len(ipArgs)
+			ipQueryParts = append(ipQueryParts, fmt.Sprintf("($%d, $%d, $%d, $%d, $%d)", argStart+1, argStart+2, argStart+3, argStart+4, argStart+5))
+			ipArgs = append(ipArgs, ip.Address, ip.Gateway, iface.Name, id, ip.ServiceAddress)
 		}
 	}
 
-	current := TOServer{}
-	err := s.ReqInfo.Tx.QueryRowx(selectV20UpdatesQuery()+` WHERE sv.id=$1`, strconv.Itoa(*s.ID)).StructScan(&current)
+	ifaceQry += strings.Join(ifaceQueryParts, ",")
+	log.Debugf("Inserting interfaces for new server, query is: %s", ifaceQry)
+
+	ifaceRows, err := tx.Query(ifaceQry, ifaceArgs...)
 	if err != nil {
 		return api.ParseDBError(err)
 	}
-	defaultIsService := true
-	if s.IPIsService == nil {
-		if current.IPIsService != nil {
-			s.IPIsService = current.IPIsService
-		} else {
-			s.IPIsService = &defaultIsService
+	defer ifaceRows.Close()
+	insertedIfaces := 0
+	for ifaceRows.Next() {
+		insertedIfaces++
+	}
+	log.Debugf("Inserted %d interfaces", insertedIfaces)
+
+	ipQry += strings.Join(ipQueryParts, ",")
+	log.Debugf("Inserting IP addresses for new server, query is: %s", ipQry)
+
+	ipRows, err := tx.Query(ipQry, ipArgs...)
```

Comment:
Similar to the previous comment, this seems like it should be an `Exec`. It's also a bit of a red flag to not iterate over all of the returned `Rows` since later queries in the same transaction will fail if there are `Rows` that haven't been `Next()`'d from a previous query in the transaction.